### PR TITLE
[IT-3230] Move agora data manager CI

### PR DESCRIPTION
**Pull request by zaro0508**
Move agora data manager from using travis CI to Github actions CI. This move also changes the update workflow to us github self-hosted runners which will be much more secure.

depends on https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/1060

### CODE REVIEW

**Review by JessterB — APPROVED** [review]

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            synapseclient[pandas,pysftp]
+      - name: Setup mongo tools - mongo-tools and mongo-shell
+        uses: boly38/action-mongo-tools@stable
+      - name: Import Synapse Data
+        run: ./import-data.sh $BRANCH $SYNAPSE_USERNAME $SYNAPSE_PASSWORD $DB_HOST $DB_USER $DB_PASS
+        env:
+          BRANCH: ${{ github.ref_name }}
+          SYNAPSE_USERNAME: ${{ secrets.SYNAPSE_USERNAME }}
+          SYNAPSE_PASSWORD: ${{ secrets.SYNAPSE_PASSWORD }}
+          DB_HOST: ${{ vars.DB_HOST }}
```
Comment (xschildw) [inline .github/workflows/main.yaml]: Aren't these env secrets?

**Comment by xschildw** [comment]
Is the DB_HOST in a public subnet? In import-data.sh, I see mongoimport connects to it directly from the runner, correct?

File: .github/workflows/main.yaml (hunk 1 above)
Comment (zaro0508) [reply .github/workflows/main.yaml]: > Aren't these env secrets?

yes, take a look at the screenshot in set of files for this PR.

**Comment by zaro0508** [comment]
> Is the DB_HOST in a public subnet? In import-data.sh, I see mongoimport connects to it directly from the runner, correct?

no, the db host is in a private subnet.  The GH runner has access to it via PR https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/1060

File: .github/workflows/main.yaml (hunk 1 above)
Comment (xschildw) [reply .github/workflows/main.yaml]: Sorry my point was should they not be referred to as ${{ secrets.DB_HOST }}?

Comment (zaro0508) [reply .github/workflows/main.yaml]: nope. the doc says to use `vars` when used in an environment context.  i have tested it and `vars` seems to work.

Comment (zaro0508) [reply .github/workflows/main.yaml]: ohh, oops.  i was testing with env vars.  sorry you are correct.  i will need to fix this

Comment (zaro0508) [reply .github/workflows/main.yaml]: ok, fixed.

**Comment by xschildw** [comment]
The GH runner has the right to make AWS API calls to setup DocumentDB. Should not mongoimport try to connect to the DB_HOST with the DB_USER/PWD creds? If so DB_HOST will resolve to a private IP address and it will not be able to find the host (I assume it works like an RDS setup).

**Comment by zaro0508** [comment]
> The GH runner has the right to make AWS API calls to setup DocumentDB. Should not mongoimport try to connect to the DB_HOST with the DB_USER/PWD creds? If so DB_HOST will resolve to a private IP address and it will not be able to find the host (I assume it works like an RDS setup).

Argg! you are correct @xschildw.  Will need to rethink this whole setup.

**Comment by zaro0508** [comment]
I was thinking that an alternative solution would be to setup AWS CLI + aws-session-manager to execute those mongo commands however it appears that AWS does not support running session manager agent on RDS instances, https://repost.aws/questions/QU9-N7fbk-SLOZvegK63XrRw/using-session-manager-to-connect-rds-without-having-ec2-instance

The only other alternative i can think of is to use GH custom runners but i’m not a fan of that solution because it requires a bit of manual configuration to setup.

**Comment by zaro0508** [comment]
Changed to use GH self hosted runners

**Review by xschildw — APPROVED** [review]
How did you setup the runners?

**Comment by zaro0508** [comment]
> How did you setup the runners?

i added some more docs around that topic but mostly just followed instructions from the github documentation.

File: README.md [hunk 2]
```diff
@@ -55,17 +56,19 @@ Context specific secrets for each environment that corresponds to a git branch (
...
 ## Self hosted runners
 
-We setup a bastian host in AWS for each environment which have access to the
-databases.  Then we associate those bastian hosts to
-[Github self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#about-self-hosted-runners).
-The update is executed from these runners.  Each runner corresponds
-to an environment which  corresponds to a git branch.  When a push happens on a
-branch (i.e. develop), the update is executed on the `agora-bastian-develop` runner
-which in turn updates the development database.
+[agora2-infra] repository deploys a bastian host in AWS for each environment which have access to
```
Comment (xschildw) [inline README.md:59]: Do we need the bastian for another reason? If not I think the runners should be on the private side.

Comment (zaro0508) [reply README.md:59]: yes, that makes sense.  will make that change in agora2-infra

Comment (zaro0508) [reply README.md:59]: https://github.com/Sage-Bionetworks/agora2-infra/pull/22

**Pull request merged**